axlearn/open_api/metrics/tool_use_execution_utils.py

```diff
@@ -0,0 +1,307 @@
+    "on",
+    "once",
+    "only",
+    "or",
+    "other",
+    "our",
+    "ours",
+    "ourselves",
+    "out",
+    "over",
+    "own",
+    "re",
+    "s",
+    "same",
+    "shan",
+    "shan't",
+    "she",
+    "she's",
+    "should",
+    "should've",
+    "shouldn",
+    "shouldn't",
+    "so",
+    "some",
+    "such",
+    "t",
+    "than",
+    "that",
+    "that'll",
+    "the",
+    "their",
+    "theirs",
+    "them",
+    "themselves",
+    "then",
+    "there",
+    "these",
+    "they",
+    "this",
+    "those",
+    "through",
+    "to",
+    "too",
+    "under",
+    "until",
+    "up",
+    "ve",
+    "very",
+    "was",
+    "wasn",
+    "wasn't",
+    "we",
+    "were",
+    "weren",
+    "weren't",
+    "what",
+    "when",
+    "where",
+    "which",
+    "while",
+    "who",
+    "whom",
+    "why",
+    "will",
+    "with",
+    "won",
+    "won't",
+    "wouldn",
+    "wouldn't",
+    "y",
+    "you",
+    "you'd",
+    "you'll",
+    "you're",
+    "you've",
+    "your",
+    "yours",
+    "yourself",
+    "yourselves",
+}
+
+
+def _string_lenient_transform(str_value: str) -> str:
+    """Performs a lenient string transformation."""
+    normalized_str_value = str_value.lower()
+    normalized_str_value = normalized_str_value.translate(str.maketrans("", "", string.punctuation))
+    words = normalized_str_value.split()
+    while len(words) > 1 and words[0] in _STOP_WORDS:
+        words = words[1:]
+    while len(words) > 1 and words[-1] in _STOP_WORDS:
+        words = words[:-1]
+    normalized_str_value = " ".join(words)
+    return normalized_str_value
+
+
+def _word_set(input_str: str) -> set[str]:
+    """Returns a set of words splitting by whitespace, newline and tab,
+    and skipping empty strings from the set."""
+    return {s for s in re.split(r"\s+", input_str) if s}
+
+
+def _match_strings_bag_of_words(pred_str: str, target_str: str, threshold: float = 1.0) -> bool:
+    """Match strings using a bag of words approach.
+
+    Args:
+        pred_str: The predicted argument string.
+        target_str: The target argument string.
+        threshold: applied to (# unique common words) / (# unique pred_str words)
+
+    Returns:
+        True if recall is higher or equal to threshold; otherwise return False.
+    """
+    if pred_str == target_str:
+        return True
+
+    assert threshold > 0, "bag of words string matching threshold must be above 0"
+    pred_word_set = _word_set(pred_str)
+    target_word_set = _word_set(target_str)
+    common_words = target_word_set.intersection(pred_word_set)
+    if len(pred_word_set) == 0:
+        return False
+    ratio = len(common_words) / len(target_word_set)
+    return ratio >= threshold
+
+
+def _is_arg_value_equal(
+    pred_arg: ValueOrListOf,
+    target_arg: ValueOrListOf,
+    check_lenient: bool,
+    bag_of_words: bool,
+) -> bool:
+    """Checks if the predicted and target arguments are equal under different checks."""
+    if check_lenient:
```

Comment:
nit -- it may be more readable to handle the other case first:
```
if not check_lenient:
    return pred_arg == target_arg

# ... handle the longer case.
```

Reply:
Done. But I am not sure if it has improved readability